src/integration/resources/common/coordinator_client.go

```diff
@@ -323,13 +327,147 @@ func (c *CoordinatorClient) DeleteNamespace(namespaceID string) error {
+	topicOpts resources.M3msgTopicOptions,
+	initRequest admin.TopicInitRequest,
+) (admin.TopicGetResponse, error) {
+	url := c.makeURL(topic.InitURL)
+	logger := c.logger.With(
+		ZapMethod("initM3msgTopic"),
+		zap.String("url", url),
+		zap.String("request", initRequest.String()),
+		zap.String("topic", fmt.Sprintf("%v", topicOpts)))
+
+	resp, err := c.makeRequest(logger, url, topic.InitHTTPMethod, &initRequest, m3msgTopicOptionsToMap(topicOpts))
+	if err != nil {
+		logger.Error("failed post", zap.Error(err))
+		return admin.TopicGetResponse{}, err
+	}
+
+	var response admin.TopicGetResponse
+	if err := toResponse(resp, &response, logger); err != nil {
+		logger.Error("failed response", zap.Error(err))
+		return admin.TopicGetResponse{}, err
+	}
+
+	logger.Info("topic initialized")
+	return response, nil
+}
+
+// GetM3msgTopic fetches an m3msg topic
+func (c *CoordinatorClient) GetM3msgTopic(
+	topicOpts resources.M3msgTopicOptions,
+) (admin.TopicGetResponse, error) {
+	url := c.makeURL(topic.GetURL)
+	logger := c.logger.With(
+		ZapMethod("getM3msgTopic"), zap.String("url", url),
+		zap.String("topic", fmt.Sprintf("%v", topicOpts)))
+
+	resp, err := c.makeRequest(logger, url, topic.GetHTTPMethod, nil, m3msgTopicOptionsToMap(topicOpts))
+	if err != nil {
+		logger.Error("failed get", zap.Error(err))
+		return admin.TopicGetResponse{}, err
+	}
+
+	var response admin.TopicGetResponse
+	if err := toResponse(resp, &response, logger); err != nil {
+		logger.Error("failed response", zap.Error(err))
+		return admin.TopicGetResponse{}, err
+	}
+
+	logger.Info("topic get")
+	return response, nil
+}
+
+//nolint:dupl
+// AddM3msgTopicConsumer adds a consumer service to an m3msg topic
+func (c *CoordinatorClient) AddM3msgTopicConsumer(
+	topicOpts resources.M3msgTopicOptions,
+	addRequest admin.TopicAddRequest,
+) (admin.TopicGetResponse, error) {
+	url := c.makeURL(topic.AddURL)
+	logger := c.logger.With(
+		ZapMethod("addM3msgTopicConsumer"),
+		zap.String("url", url),
+		zap.String("request", addRequest.String()),
+		zap.String("topic", fmt.Sprintf("%v", topicOpts)))
+
+	resp, err := c.makeRequest(logger, url, topic.AddHTTPMethod, &addRequest, m3msgTopicOptionsToMap(topicOpts))
+	if err != nil {
+		logger.Error("failed post", zap.Error(err))
+		return admin.TopicGetResponse{}, err
+	}
+
+	var response admin.TopicGetResponse
+	if err := toResponse(resp, &response, logger); err != nil {
+		logger.Error("failed response", zap.Error(err))
+		return admin.TopicGetResponse{}, err
+	}
+
+	logger.Info("topic consumer added")
+	return response, nil
+}
+
+func m3msgTopicOptionsToMap(opts resources.M3msgTopicOptions) map[string]string {
+	return map[string]string{
+		headers.HeaderClusterEnvironmentName: opts.Env,
+		headers.HeaderClusterZoneName:        opts.Zone,
+		topic.HeaderTopicName:                opts.TopicName,
+	}
+}
+
+// GetAggPlacement gets aggregator placements.
+func (c *CoordinatorClient) GetAggPlacement() (admin.PlacementGetResponse, error) {
```

Comment:
We should modify the existing placement methods on the interface to take a placement type (e.g. aggregator, dbnode) instead of having separate methods. I'm pretty sure the only distinction is the URL that's used. We can choose that URL based on the placement type.

Reply:
yes, I had a comment about merging GetPlacement and GetAggPlacment somewhere. Just added the PlacementRequestOptions to indicate the service type.